## 어떻게 해야 mvp상을 받을 수 있을까?
- 2020~2023년 메이저리그 mvp상을 받은 인물의 특징을 분석
- mvp 선수의 특징과 전체선수의 특징을 비교
- 그리고 mvp선수의 어떤점이 특출 났길래 mvp상을 받을 수 있었는지 비교 분석하기
- 가장 뛰어난 선수한명을 소개하기


In [ ]:
# 모듈 로딩
import pandas as pd
import numpy as np

In [ ]:
# 데이터 준비
# 2021~2023년도 현역으로 뛰었던 선수들의 데이터
# 2020년도 이하에는 코로나로 인해 경기수가 적어서 데이터가 부족하기에
# 좀더 정확한 평균치와 간편한 비교를 위해 2021년도 자료부터 조사했습니다.
file_name='stats.tsv'
yagu=pd.read_csv(file_name, sep='\t')
yagu

년도   타석  홈런   삼진  볼넷  삼진비율  볼넷비율     타율    출루율  \
1   Semien, Marcus         2023  753  29  110  72  14.6   9.6  0.276  0.826   
2   Acuña Jr., Ronald      2023  735  41   84  80  11.4  10.9  0.337  1.012   
3   Freeman, Freddie       2023  730  29  121  72  16.6   9.9  0.331  0.977   
4   Bregman, Alex          2023  724  25   87  92  12.0  12.7  0.262  0.804   
5   Lowe, Nathaniel        2023  724  17  165  93  22.8  12.8  0.262  0.774   
...                         ...  ...  ..  ...  ..   ...   ...    ...    ...   
534 Vázquez, Christian     2020  189   7   43  16  22.8   8.5  0.283  0.801   
535 Arcia, Orlando         2020  189   5   32  14  16.9   7.4  0.260  0.733   
536 Robles, Victor         2020  189   3   53   9  28.0   4.8  0.220  0.608   
537 Brantley Jr., Michael  2020  187   5   28  17  15.0   9.1  0.300  0.840   
538 Dozier, Hunter         2020  186   6   48  27  25.8  14.5  0.228  0.736   

                            타점  예상장타율  가중출루율  예상출루율  헛스윙비율  스윙비율  
1   Semien, Marcus         100  0.427  0.354  0.337   18.0  47.1  
2   Acuña Jr., Ronald      106  0.660  0.428  0.435   18.7  46.2  
3   Freeman, Freddie       102  0.565  0.411  0.403   20.4  51.0  
4   Bregman, Alex           98  0.430  0.351  0.368   14.1  38.7  
5   Lowe, Nathaniel         82  0.406  0.340  0.362   20.6  43.8  
...                        ...    ...    ...    ...    ...   ...  
534 Vázquez, Christian      23  0.358  0.346  0.289   22.9  47.6  
535 Arcia, Orlando          20  0.428  0.317  0.339   23.1  46.8  
536 Robles, Victor          15  0.280  0.273  0.284   27.1  50.3  
537 Brantley Jr., Michael   22  0.413  0.362  0.327   16.3  39.9  
538 Dozier, Hunter          12  0.385  0.326  0.351   30.9  41.9  

[538 rows x 15 columns]

In [ ]:
# 2023년, 2022년, 2021년 mvp 선수의 자료를 추출하기
# 내셔널 리그 mvp 선수 출력
yagu.iloc[1] # 2023
yagu.iloc[161] # 2022
yagu.iloc[324] # 2021


년도       2021.000
타석        599.000
홈런         35.000
삼진        134.000
볼넷        100.000
삼진비율       22.400
볼넷비율       16.700
타율          0.309
출루율         1.044
타점         84.000
예상장타율       0.609
가중출루율       0.431
예상출루율       0.423
헛스윙비율      31.900
스윙비율       47.500
Name: (325,  Harper, Bryce), dtype: float64

In [ ]:
# 아메리칸 리그 mvp 선수 출력
yagu.iloc[74] # 2023 yagu.iloc[1] # 2023
yagu.iloc[141] # 2022 yagu.iloc[161] # 2022
yagu.iloc[300] # 2021 yagu.iloc[324] # 2021

년도       2021.000
타석        639.000
홈런         46.000
삼진        189.000
볼넷         96.000
삼진비율       29.600
볼넷비율       15.000
타율          0.257
출루율         0.964
타점        100.000
예상장타율       0.612
가중출루율       0.393
예상출루율       0.382
헛스윙비율      35.000
스윙비율       45.800
Name: (301,  Ohtani, Shohei), dtype: float64

In [ ]:
# 네셔널 리그 mvp선수 데이터 프레임으로 만들기
national23=pd.DataFrame(yagu.iloc[1])
national22=pd.DataFrame(yagu.iloc[161])
national21=pd.DataFrame(yagu.iloc[324])

# 아메리칸리그 mvp선수 데이터프레임으로 만들기
american23=pd.DataFrame(yagu.iloc[74])
american22=pd.DataFrame(yagu.iloc[141])
american21=pd.DataFrame(yagu.iloc[300])

In [ ]:
# DataFrame을 년도 별로 합치기
mvp2023 = pd.concat([national23, american23], axis=1)
mvp2023


2              75
      Acuña Jr., Ronald Ohtani, Shohei
년도             2023.000       2023.000
타석              735.000        599.000
홈런               41.000         44.000
삼진               84.000        143.000
볼넷               80.000         91.000
삼진비율             11.400         23.900
볼넷비율             10.900         15.200
타율                0.337          0.304
출루율               1.012          1.066
타점              106.000         95.000
예상장타율             0.660          0.638
가중출루율             0.428          0.433
예상출루율             0.435          0.406
헛스윙비율            18.700         32.300
스윙비율             46.200         48.200

In [ ]:
mvp2022 = pd.concat([national22, american22], axis=1)
mvp2022

162          142
      Goldschmidt, Paul Judge, Aaron
년도             2022.000     2022.000
타석              651.000      696.000
홈런               35.000       62.000
삼진              141.000      175.000
볼넷               79.000      111.000
삼진비율             21.700       25.100
볼넷비율             12.100       15.900
타율                0.317        0.311
출루율               0.982        1.111
타점              115.000      131.000
예상장타율             0.482        0.706
가중출루율             0.419        0.458
예상출루율             0.357        0.422
헛스윙비율            26.800       29.800
스윙비율             41.700       42.700

In [ ]:
mvp2021 = pd.concat([national21, american21], axis=1)
mvp2021

325            301
      Harper, Bryce Ohtani, Shohei
년도         2021.000       2021.000
타석          599.000        639.000
홈런           35.000         46.000
삼진          134.000        189.000
볼넷          100.000         96.000
삼진비율         22.400         29.600
볼넷비율         16.700         15.000
타율            0.309          0.257
출루율           1.044          0.964
타점           84.000        100.000
예상장타율         0.609          0.612
가중출루율         0.431          0.393
예상출루율         0.423          0.382
헛스윙비율        31.900         35.000
스윙비율         47.500         45.800

In [ ]:
# 모든년도 DataFrame을 하나로 만들기
a=pd.concat([mvp2023, mvp2022], axis=1)
a

2              75                162          142
      Acuña Jr., Ronald Ohtani, Shohei Goldschmidt, Paul Judge, Aaron
년도             2023.000       2023.000          2022.000     2022.000
타석              735.000        599.000           651.000      696.000
홈런               41.000         44.000            35.000       62.000
삼진               84.000        143.000           141.000      175.000
볼넷               80.000         91.000            79.000      111.000
삼진비율             11.400         23.900            21.700       25.100
볼넷비율             10.900         15.200            12.100       15.900
타율                0.337          0.304             0.317        0.311
출루율               1.012          1.066             0.982        1.111
타점              106.000         95.000           115.000      131.000
예상장타율             0.660          0.638             0.482        0.706
가중출루율             0.428          0.433             0.419        0.458
예상출루율             0.435          0.406             0.357        0.422
헛스윙비율            18.700         32.300            26.800       29.800
스윙비율             46.200         48.200            41.700       42.700

In [ ]:
# 모든년도 DataFrame을 하나로 만들기
b=pd.concat([mvp2021], axis=1)
b

325            301
      Harper, Bryce Ohtani, Shohei
년도         2021.000       2021.000
타석          599.000        639.000
홈런           35.000         46.000
삼진          134.000        189.000
볼넷          100.000         96.000
삼진비율         22.400         29.600
볼넷비율         16.700         15.000
타율            0.309          0.257
출루율           1.044          0.964
타점           84.000        100.000
예상장타율         0.609          0.612
가중출루율         0.431          0.393
예상출루율         0.423          0.382
헛스윙비율        31.900         35.000
스윙비율         47.500         45.800

In [ ]:
# 2021~2023 mvp 선수 DataFrame
mvpman=pd.concat([a,b], axis=1)
mvpman

2              75                162          142  \
      Acuña Jr., Ronald Ohtani, Shohei Goldschmidt, Paul Judge, Aaron   
년도             2023.000       2023.000          2022.000     2022.000   
타석              735.000        599.000           651.000      696.000   
홈런               41.000         44.000            35.000       62.000   
삼진               84.000        143.000           141.000      175.000   
볼넷               80.000         91.000            79.000      111.000   
삼진비율             11.400         23.900            21.700       25.100   
볼넷비율             10.900         15.200            12.100       15.900   
타율                0.337          0.304             0.317        0.311   
출루율               1.012          1.066             0.982        1.111   
타점              106.000         95.000           115.000      131.000   
예상장타율             0.660          0.638             0.482        0.706   
가중출루율             0.428          0.433             0.419        0.458   
예상출루율             0.435          0.406             0.357        0.422   
헛스윙비율            18.700         32.300            26.800       29.800   
스윙비율             46.200         48.200            41.700       42.700   

                325            301  
      Harper, Bryce Ohtani, Shohei  
년도         2021.000       2021.000  
타석          599.000        639.000  
홈런           35.000         46.000  
삼진          134.000        189.000  
볼넷          100.000         96.000  
삼진비율         22.400         29.600  
볼넷비율         16.700         15.000  
타율            0.309          0.257  
출루율           1.044          0.964  
타점           84.000        100.000  
예상장타율         0.609          0.612  
가중출루율         0.431          0.393  
예상출루율         0.423          0.382  
헛스윙비율        31.900         35.000  
스윙비율         47.500         45.800

In [ ]:
# column을 변경.
mvpman.columns=['Acuña Jr., Ronald(nat23)','Ohtani, Shohei(amer23)','Goldschmidt, Paul(nat22)','Judge, Aaron(amer22)','Harper, Bryce(nat21)','Ohtani, Shohei(amer21)']

In [ ]:
mvpman

Acuña Jr., Ronald(nat23)  Ohtani, Shohei(amer23)  \
년도                     2023.000                2023.000   
타석                      735.000                 599.000   
홈런                       41.000                  44.000   
삼진                       84.000                 143.000   
볼넷                       80.000                  91.000   
삼진비율                     11.400                  23.900   
볼넷비율                     10.900                  15.200   
타율                        0.337                   0.304   
출루율                       1.012                   1.066   
타점                      106.000                  95.000   
예상장타율                     0.660                   0.638   
가중출루율                     0.428                   0.433   
예상출루율                     0.435                   0.406   
헛스윙비율                    18.700                  32.300   
스윙비율                     46.200                  48.200   

       Goldschmidt, Paul(nat22)  Judge, Aaron(amer22)  Harper, Bryce(nat21)  \
년도                     2022.000              2022.000              2021.000   
타석                      651.000               696.000               599.000   
홈런                       35.000                62.000                35.000   
삼진                      141.000               175.000               134.000   
볼넷                       79.000               111.000               100.000   
삼진비율                     21.700                25.100                22.400   
볼넷비율                     12.100                15.900                16.700   
타율                        0.317                 0.311                 0.309   
출루율                       0.982                 1.111                 1.044   
타점                      115.000               131.000                84.000   
예상장타율                     0.482                 0.706                 0.609   
가중출루율                     0.419                 0.458                 0.431   
예상출루율                     0.357                 0.422                 0.423   
헛스윙비율                    26.800                29.800                31.900   
스윙비율                     41.700                42.700                47.500   

       Ohtani, Shohei(amer21)  
년도                   2021.000  
타석                    639.000  
홈런                     46.000  
삼진                    189.000  
볼넷                     96.000  
삼진비율                   29.600  
볼넷비율                   15.000  
타율                      0.257  
출루율                     0.964  
타점                    100.000  
예상장타율                   0.612  
가중출루율                   0.393  
예상출루율                   0.382  
헛스윙비율                  35.000  
스윙비율                   45.800

In [ ]:
# 2021~2023년도 mvp선수들의 평균 DataFrame
# 일단 내셔널 리그와 아메리칸 리그는 경기수도 같기 때문에 같은 리그라고 생각했습니다.
# 내셔널 리그와 아메리칸 리그의 큰 차이점은 지명투수의 유무입니다.
# 내셔널 리그는 지명투수가 없어서 투수가 벤츠에 들어가거나 다시 나오는 경우가 있다고 합니다.
mvp_mean=mvpman.mean(axis=1)
mvp_mean=pd.DataFrame(mvp_mean)
mvp_mean.columns=['2021~2023 mvp선수 평균 특징']
mvp_mean=mvp_mean.drop(['년도'])



In [ ]:
# yagu DataFrame 평균구하기
yagu=yagu.T
yagu_mean=yagu.mean(axis=1)
yagu_mean=pd.DataFrame(yagu_mean)
yagu_mean.columns=['2021~2023 전체선수 평균 특징']
yagu_mean=yagu_mean.drop(['년도'])

In [ ]:
# 전체 야구선수 평균과 mvp 선수 평균 합치기
total=pd.concat([mvp_mean, yagu_mean], axis=1)
total

2021~2023 mvp선수 평균 특징  2021~2023 전체선수 평균 특징
타석                653.166667            501.771375
홈런                 43.833333             18.347584
삼진                144.333333            103.282528
볼넷                 92.833333             45.700743
삼진비율               22.350000             20.827138
볼넷비율               14.300000              9.154089
타율                  0.305833              0.261448
출루율                 1.029833              0.783245
타점                105.166667             62.665428
예상장타율               0.617833              0.440056
가중출루율               0.427000              0.337136
예상출루율               0.404167              0.334461
헛스윙비율              29.083333             24.382342
스윙비율               45.350000             47.169888

MVP와의 관련성  

MVP 선정 시 다양한 요소가 고려되며, 위의 통계들은 선수의 성과를 종합적으로 평가하는 데 중요한 역할을 합니다.
지금부터 mvp 선수들 평균을 사람한명의 성적으로 칭하겠습니다. 이 사람의 이름은 A 군
전체평균 선수들의 평균을 사람한명의 성적으로 칭하겠습니다. 이 사람의 이름은 B 군
지금 부터 A군과 B군의 실력을 비교하겠습니다.

1.타석(PA), 홈런(HR), 타점(RBI):
  타석 수, 홈런 수, 타점은 선수의 공격력을 직접적으로 나타내며, MVP 선정에 큰 영향을 미칩니다.
  결과를 보시면 A군은 B군 보다 타석수가 152타 더 많습니다.
  홈런차이는 약 2.3배나 차이가 납니다.
  타점은 내가 안타를 쳤을 때 1년기준으로 홈으로 들어오는 아군의 숫자를 말합니다.
  A군의 타점은 1년에 105점을 얻을 수 있는 반면 B 군은 62점을 얻습니다.
  무려 43점을 더 얻을 수 있습니다.


2.삼진(SO), 삼진 비율(K%), 헛스윙 비율(Whiff %):
  삼진과 헛스윙 비율은 타자의 약점을 나타내지만, 높은 삼진율에도 불구하고 강한 공격력을 보여주는 선수는 MVP 후보로 선정될 수 있습니다.
  삼진 같은 경우에는 A군이 B군보다 높습니다. A군은 1년에 144번 삼진하는 반면 B군은 103번 삼진을 합니다.
  삼진 확률도 A군은 22.35%, B군은 20.8%로 A군이 더 높습니다.
  헛스윙 비율도 B군이 24.3%고 A군은 29%로 B군이 5% 앞서는 모습입니다.
  아무래도 이 데이터는 mvp상을 받기 위한 영향이 그리 크지 않은거 같습니다.  

3.볼넷(BB), 볼넷 비율(BB%), 출루율(OPS, xOBP):
  볼넷 수와 출루율은 타자의 선구안과 출루 능력을 나타내며, 팀에 중요한 기여를 하는 요소로 평가됩니다.
  A군은 볼넷으로 인한 출루가 92번인 반면 B군은 45번에 그쳤습니다.
  볼넷 비율도 A군은 14%, B군은 9%로 5%앞선 모습입니다.
  제가 조사한 데이터에서는 예상출루율과 가중출루율이 있는데. 출루율은 안타, 볼넷, 몸에 맞는 볼 등을 포함하여
  타자가 출루할 수 있는 모든 경우를 고려하여 계산한 확률입니다.
  요즘 추세는 예상출루율보다 가중출루율을 높게 평가합니다.
  가중출루율은 실력으로 계산한 확률이기 때문에 정확도가 더 높기 때문에 가중출루율로 설명하겠습니다.
  A군은 42%, B군은 33%로 11% 높은 출루율을 보여줍니다.
  출루율은 안타, 볼넷, 데스볼 3가지 요소를 따져서 만든 확율이기 때문에 1%,2% 차이가 많은 기량차이로 이어집니다.  

4.타율(AVG), 출루율 + 장타율(OPS)
  타율, 출루율 + 장타율, 가중 출루율은 타자의 전반적인 타격 능력을 평가하는 데 중요한 지표입니다.
  타율 같은 경우에는 A군이 30%, B군이 24%대를 보입니다. 보통 3할 2할이라고 합니다.
  1할차이는 야구에서는 큰 변수를 만들어 내느냐 마느냐를 가름할 수 있기 때문에 0.6할 차이는 많은 차이가 난다고 할 수 있습니다.
  장타율은 공을 쳤을 때 2루타 이상 공이 나갔을 경우를 장타라고 합니다.
  위의 ops 즉, 예상장타율이라 정리한 부분은 출루율과 장타율을 합쳤을 때 값입니다.
  A군의 ops는 60%, B군의 ops는 44% 대로 16%차이가 납니다.
  이또한 두종류의 확률을 합친 확률이기 때문에 16%는 많은 기량차이를 나타낸다 볼 수 있겠습니다.

6.스윙 비율(Swing %):
스윙 비율은 타자의 공격 성향을 나타냅니다. 공격적으로 스윙하는 타자가 많은 안타와 홈런을 기록할 수 있습니다.
A군은 45%, B군은 47% 차이로 2% 차이밖에 나지 않아 MVP상에 영향이 그리 크지 않은거 같습니다.

종합적으로 mvp상의 중요한 요소입니다.
A군은 타석, 홈런, 타점 부분에서 B군과 많은 차이를 보여줍니다.
다음요소는 볼넷, 볼넷 비율, 출루율에서 A군이 많은 차이를 보여주었구요.
마지막 요소로는 타율, ops 부분에서 A군이 많은 차이를 보여주었습니다.

결론적으로 mvp상에 많은 영향을 미치는 요소는 무엇인가 정리해보니.
1. 타석, 홈런, 타점
2. 볼넷, 볼넷 비율, 출루율
3. 타율, ops
의 기량이 높은 선수가 mvp상을 쟁취 할 수 있는 모습입니다.



그래서 올해 mvp상 강력한 우승 후보를 찾아보았습니다.  

바로 오타니 쇼헤이 선수인데요. 아까 A군이 오타니 쇼헤이 선수라 생각하시면 될 것 같습니다.  

이선수는 아메리칸리그에서 2021년 2023년 2번이나 mvp상을 수여 받은 선수입니다.  

https://baseballsavant.mlb.com/savant-player/shohei-ohtani-660271?stats=statcast-r-hitting-mlb  

오타니 선수의 홈런 영상을 시청해 보시겠습니다.  

https://www.mlb.com/video/data-viz-shohei-ohtani-s-400-foot-home-run?partnerId=web_video-playback-page_video-share



